### PR TITLE
Dockerfile: tuning, tuning, tuning

### PR DESCRIPTION
**Pull request by bryanlatten**
- Added a number of tests, too.

### CODE REVIEW

File: container/root/etc/nginx/nginx.conf [hunk 1]
```diff
@@ -16,12 +16,66 @@ pid /tmp/.nginx/nginx.pid;
 
 error_log /dev/stdout warn;
 
+# Number of file descriptors used for nginx
+worker_rlimit_nofile 40000;
```
Comment (ko-be) [inline container/root/etc/nginx/nginx.conf]: Should these parameters be configurable as well @bryanlatten ?

Comment (bryanlatten) [inline container/root/etc/nginx/nginx.conf]: in theory, yes. but I can't think of much of a reason to modify it at runtime. instead, a downstream client would simply `sed` it out

**Comment by ko-be** [comment]
👍

File: container/root/tests/nginx/base.goss.yaml [hunk 2]
```diff
@@ -1,12 +1,63 @@
...
+      - client_max_body_size 1m;
+      - try_files $uri $uri/ =404;
+      - location ~ /\. {
+      - access_log off;
+      - log_not_found off;
+      - deny all;
+  /etc/nginx/nginx.conf:
+    exists: true
+    contains:
+      - '/;{0}worker_processes\s*(auto|\d+);/'
```
Comment (digitalyuki) [inline container/root/tests/nginx/base.goss.yaml]: @bryanlatten How do these goss tests work?

Comment (bryanlatten) [inline container/root/tests/nginx/base.goss.yaml]: regex :)

Comment (digitalyuki) [inline container/root/tests/nginx/base.goss.yaml]: @bryanlatten , can you explain one of the example regex's? 
i'm interpreting, say, this line as:
contains a line with "zero semicolons to start", then `worker_processes`, whitespace(s) then `auto` or one or more digits. 

Why is the first `;{0}` required? that part makes no sense to me.

**Comment by apratina** [comment]
👍

File: container/root/tests/nginx/base.goss.yaml (hunk 2 above)
Comment (bryanlatten) [inline container/root/tests/nginx/base.goss.yaml]: @digitalyuki ensuring it isn't commented out

Comment (bryanlatten) [inline container/root/tests/nginx/base.goss.yaml]: @digitalyuki this is why we peer review :) great catch

**Comment by digitalyuki** [comment]
👍

**Comment by mannytoledo** [comment]
very belated 👍

**Pull request merged**